File: Deepseek-Coder-33b-Instruct/stage2/serial_construct.c
#include <stdlib.h>
#include <time.h>
#include "acc_testsuite.h"

#ifndef T1
//T1:serial construct,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    const int size = 100;
    int data[size];
    long long int sum = 0;
    long long int expected_sum = 0;

    // Initialize the array with random numbers
    for(int i = 0; i < size; i++){
        data[i] = rand() % 100;
        expected_sum += data[i];
    }

    // Use the serial construct to sum the array
    #pragma acc serial
    {
        for(int i = 0; i < size; i++){
            sum += data[i];
        }
    }

    // Check if the sum is correct
    if(sum != expected_sum){
        err++;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/serial_construct.f90
!T1:serial construct,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER :: a(10), b(10), c(10)
        INTEGER :: i

        ! Initialize arrays
        DO i = 1, 10
          a(i) = i
          b(i) = i * 2
          c(i) = 0
        END DO

        !$acc parallel loop
        DO i = 1, 10
          c(i) = a(i) + b(i)
        END DO

        !$acc serial
        DO i = 1, 10
          IF (c(i) .ne. a(i) + b(i)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM